### DISCUSSION TITLE
Dark mode contrast

### DISCUSSION THREAD

**Discussion by hamiltlr** [firefly-iii/firefly-iii#7141] [General]
I just updated to version 6.0.0, and it looks like a nice release overall!
I do unfortunately have a big gripe relating to the new dark mode. I am visually impaired, and normally I use dark mode wherever possible as it's easier for me to see. However the new dark mode in Firefly just doesn't have enough contrast for me, and makes it harder to see. A lot of the text for example is just too similar to the background that I have to strain to see it, especially on mobile where the text is smaller. Any text that's close to white colored is generally fine, but for example the colored text on the light grey background in the budgets view is very difficult for me to see:
![image](https://user-images.githubusercontent.com/1261920/222970104-0b43bf70-986b-49de-be11-f273c28d9f78.png)
The red and green here is difficult to see because there's not much contrast between it and the background. 

Another example is under Asset Accounts, the amounts are a little hard for me to read, although it's better than the budgets example:
![image](https://user-images.githubusercontent.com/1261920/222970258-434a8fe8-4bb1-4b3a-a30b-1b3fcb962eaa.png)

And the sidebar headings (Financial Control, Accounting, etc) are hard to read:
![image](https://user-images.githubusercontent.com/1261920/222970549-056324c8-bb37-4f37-a148-09da39e4bd62.png)

Personally, I would love to have an "AMOLED" or high-contrast dark mode where the background is pure black.

Previously I had been using things like the Dark Reader browser extension to make the page dark, and that still mostly works on desktop. However on mobile I use the Chrome "Auto Dark Mode for Web Contents" setting, and that doesn't seem to do anything now with the built-in dark mode.
There isn't a way to go back to light mode is there? I didn't see an option for it, but if there is I could just switch back to light mode and use the same client-side methods I'm currently using.

**Comment by JC5** [comment]
Thanks for the feedback! The dark mode is based on the browser's preference. I'll update the colors for the next patch.

**Reply by Kline-** [reply]
> The dark mode is based on the browser's preference.

I suspected this but thanks for confirming since I didn't see it mentioned in the docs or anywhere else yet.

Are there any plans to make a profile setting to manually override the theme instead of the browser's preference? Similarly to the OP I generally prefer dark mode when available but the current dark mode isn't quite on par with the light mode for readability yet.

**Comment by daften** [comment]
Hey, after the changes it is better, but contrast ratio is still a bit too low (IMO). Is it possible to do wcag2aa checks for contrasts on color changes? To give one example, the contrast for negative amounts is too low for that now. ##a94442 on #272c30 only gets a contrast score of 2.41, while ideally it should be 4.5 minimum. If you'd like, I can try to work on it a bit when I have some time, but it would mean the overall look-and-feel would be less muted as now. What do you think about potential changes like this?

**Reply by JC5** [reply]
Go ahead!

**Comment by mindaugas-j** [comment]
would be cool to have a default/dark/light mode setting in Options->Preferences

**Comment by AvengerDr** [comment]
So at the moment there is no way to revert to the light theme? Not even by using some kind of environment variable?
Because I too like dark themes in general, but the one currently used seems too muted. 

For example, the github dark theme could be used as an inspiration as it seems to me it has more distinguishable colors.

**Reply by JC5** [reply]
Right now, use your browser settings.
